.github/workflows/checks.yml

```diff
@@ -0,0 +1,63 @@
+name: CI
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/checks.yaml'
+      - '**.go'
+
+permissions:
+  # Permission for checking out code
+  contents: read
+
+jobs:
+  build:
+    name: Check & Build Provider
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - run: chmod +x ./scripts/gofmtcheck.sh
+      - name: gofmt Check
+        run: ./scripts/gofmtcheck.sh
+      # - name: Generate provider code
+      #   run: go generate
+      # - name: Check generated code for diffs
+      #   run: git diff --exit-code
+      - name: Build
+        run: go build -v
+  acceptance:
+    name: Acceptance Tests
+    needs: build
+    runs-on: ubuntu-latest
+    concurrency:
+        group: tf-aci-ci-test-${{ matrix.apic_host.name }}
+        cancel-in-progress: false
+    strategy:
+      fail-fast: false
+      matrix:
+        apic_host:
+          - name: v5.2
+            url: 'https://173.36.219.83/'
+          - name: v6.0
+            url: 'https://173.36.219.84/'
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: '1.7.*'
+          terraform_wrapper: false
+      - name: Terraform Acceptance Test (APIC ${{ matrix.apic_host.name }})
+        run: go test github.com/CiscoDevNet/terraform-provider-aci/v2/internal/provider -v -cover -timeout 120m
+        env:
+          TF_ACC: '1'
+          TF_ACC_STATE_LINEAGE: '1'
+          ACI_VAL_REL_DN: false
+          ACI_USERNAME: ${{ secrets.TF_ACC_ACI_USERNAME }}
+          ACI_PASSWORD: ${{ secrets.TF_ACC_ACI_PASSWORD }}

```

Comment:
My understanding is that secrets cannot be used when using PR from forks. They will use the forks value for that secret not the main repo.

Reply:
I plan to remove these secrets and reuse the hosts/username/password from the ansible-aci repo after this is merged.